File: frontend/src/model/model/entity/Poi.ts
import BaseEntity from '../base/BaseEntity.ts'

/**
 * 兴趣点
 */
export default class Poi extends BaseEntity {
  /**
   * 主键
   */
  id: number | undefined | null
  /**
   * 兴趣点名称
   */
  poiName: string | undefined | null
  constructor(poi: Poi) {
    super(poi)
    this.id = poi.id
    this.poiName = poi.poiName
  }
}

File: backend/base/model/entity/poi.go
package entity

import (
	"database/sql"

	"github.com/library-squirrel/wails/backend/base/model"
)

// Poi POI
type Poi struct {
	*model.BaseEntity                // 嵌入基础实体
	PoiName           sql.NullString `gorm:"column:poi_name" json:"poiName"`
}

func NewPoi() *Poi {
	return &Poi{
		BaseEntity: &model.BaseEntity{},
	}
}

func (Poi) TableName() string {
	return "poi"
}
